assets/src/dashboard/components/previewPage.js

```diff
@@ -13,21 +13,59 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+/**
+ * External dependencies
+ */
+import { useEffect } from 'react';
+import PropTypes from 'prop-types';
 
 /**
  * Internal dependencies
  */
 import DisplayElement from '../../edit-story/components/canvas/displayElement';
 import StoryPropTypes from '../../edit-story/types';
+import StoryAnimation, { useStoryAnimationContext } from './storyAnimation';
+
+function PreviewPageController({ page, animationState }) {
+  const {
+    actions: { playWAAPIAnimations },
+  } = useStoryAnimationContext();
+
+  useEffect(() => {
+    if ('animate' === animationState) {
+      playWAAPIAnimations();
+    }
+  }, [animationState, playWAAPIAnimations]);
 
-function PreviewPage({ page }) {
   return page.elements.map(({ id, ...rest }) => (
-    <DisplayElement key={id} page={page} element={{ id, ...rest }} />
+    <DisplayElement
+      key={id}
+      page={page}
+      element={{ id, ...rest }}
+      animationMode="WAAPI"
+    />
   ));
 }
 
+function PreviewPage({ page, animationState = 'idle', onAnimationComplete }) {
+  return (
+    <StoryAnimation.Provider
+      animations={page.animations}
+      onWAAPIFinish={onAnimationComplete}
+    >
+      <PreviewPageController
+        page={page}
+        animationState={animationState}
+        onAnimationComplete={onAnimationComplete}
+      />
+    </StoryAnimation.Provider>
+  );
+}
+
 PreviewPage.propTypes = {
   page: StoryPropTypes.page.isRequired,
+  animationState: PropTypes.oneOf(['animate', 'idle']),
```

Comment:
If you update `animate` and `idle` in the state machine for preview you can reference it here too.